### PR TITLE
Add `playpass.device.getBestShareType()`.

### PR DESCRIPTION
**Pull request by aduros**
I'm not too crazy on "device" being the namespace name, but the idea is
that we might add other device detection functions here someday.

### CODE REVIEW

File: src/device/index.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    {
+        shareType: ShareType.Twitter,
+
+        // No userAgent detection available on Twitter!
+
+        // twitter.com seems unused, but included here for safety
+        referrer: /\b(t\.co|twitter\.com)$/,
+    },
+
+    // TODO(2022-06-23): Add Reddit?
```
Comment (joshplayco) [inline src/device/index.ts:35]: yes pls

File: src/device/index.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+export function getBestShareType (): ShareType {
+    const referrer = document.referrer && new URL(document.referrer).hostname;
+
+    for (const detector of detectors) {
+        if (detector.userAgent?.test(navigator.userAgent) || detector.referrer?.test(referrer)) {
+            return detector.shareType;
+        }
+    }
+
+    return ShareType.Any;
```
Comment (joshplayco) [inline src/device/index.ts:54]: so if we were creating one button, the default return here would indicate to just open the native share sheet (or desktop)?

Comment (aduros) [reply src/device/index.ts:54]: Yup, exactly.

File: src/device/index.ts (hunk 1 above)
Comment (nhydock) [reply src/device/index.ts:35]: reddit referrer should be `/b(redd.it|reddit\.com)$/` as far as I can tell

**Pull request merged**